### PR TITLE
Add more modules

### PR DESCRIPTION
**Pull request by chimp1984**
Based on https://github.com/bisq-network/misq/pull/5

Adds modules as they are used in the prototype code base. Most do not have an implementation yet but define the structure and dependency graph. 

Security module has working implementation and will require a review from a cryptographer (@stejbac would be great if you can have a look over it).

### CODE REVIEW

File: security/src/main/java/network/misq/security/HybridEncryption.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.PublicKey;
+import java.security.Security;
+import java.util.Arrays;
+
+import static network.misq.common.util.ByteArrayUtils.concat;
```
Comment (ghost) [inline security/src/main/java/network/misq/security/HybridEncryption.java:33]: Missing import
```suggestion
import static com.google.common.base.Preconditions.checkArgument;
import static network.misq.common.util.ByteArrayUtils.concat;
```

File: contract/src/main/java/network/misq/contract/Party.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package network.misq.contract;
+
+import lombok.Getter;
+import network.misq.network.p2p.NetworkId;
```
Comment (ghost) [inline contract/src/main/java/network/misq/contract/Party.java:21]: Build complains `Party.java:21: error: package network.misq.network.p2p does not exist`

**Review by jmacxx — CHANGES_REQUESTED** [review]

File: contract/src/main/java/network/misq/contract/Party.java (hunk 2 above)
Comment (chimp1984) [reply contract/src/main/java/network/misq/contract/Party.java:21]: I will have a look. I merge it with the changes of the other PRs.

Comment (chimp1984) [reply contract/src/main/java/network/misq/contract/Party.java:21]: Hm... for me it works. Made a clean and full build. In IntelliJ there are issues when you add modules that it does not refelct that in the IDE. Easiest is to shutdown, delete .idea folder and let it rebuild it from the gradle module definitions.

Comment (chimp1984) [reply contract/src/main/java/network/misq/contract/Party.java:21]: Merged other PRs and rebased....

Comment (chimp1984) [reply contract/src/main/java/network/misq/contract/Party.java:21]: > Build complains `Party.java:21: error: package network.misq.network.p2p does not exist`

Ah seems there was a commit not pushed ;-(... Should work now.

**Comment by chimp1984** [comment]
Merged other PRs and rebased....

**Review by jmacxx — APPROVED** [review]

File: security/src/main/java/network/misq/security/ConfidentialData.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        this.hmac = hmac;
+        this.iv = iv;
+        this.cypherText = cypherText;
+        this.signature = signature;
+    }
+
+    @Override
+    public String toString() {
+        return "Sealed{" +
+                ",\r\n     hmac=" + Hex.encode(hmac) +
```
Comment (kaiwolfram) [inline security/src/main/java/network/misq/security/ConfidentialData.java]: It's the first value
```suggestion
                "\r\n     hmac=" + Hex.encode(hmac) +
```

**Comment by ghubstan** [comment]
utACK

**Pull request merged**